### PR TITLE
Close logger's internal timer to avoid busy event loop

### PR DESCRIPTION
**Pull request by adrukh**
Adding the `close` method to clear `_sendTimer` and purge any pending logs. This frees up the event loop for test scenarios where the logger would prevent the test from ever completing.

@gillyb this should later propagate into winston-logzio, I'll issue another PR once this is approved.

Similar to @barakhaim 's pull request, with tests and preventing additional logs into closed logger.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,7 +27,9 @@
   "dependencies": {
     "json-stringify-safe": "5.0.x",
     "lodash": "^3.10.1",
-    "request": "2.49.x"
+    "request": "2.49.x",
+    "sequelize": "2.1.3",
```
Comment (asafm) [inline package.json]: Where are you using this?

File: package.json [hunk 2]
```diff
@@ -27,7 +27,9 @@
...
+    "console-io": "2.2.13"
```
Comment (asafm) [inline package.json]: Same here

File: package.json [hunk 3]
```diff
@@ -27,7 +27,9 @@
   "dependencies": {
     "json-stringify-safe": "5.0.x",
     "lodash": "^3.10.1",
-    "request": "2.49.x"
+    "request": "2.49.x",
```
Comment (asafm) [inline package.json]: Where are you using this?

**Comment by adrukh** [comment]
@asafm sorry for the hassle, did some local tinkering on my fork and forgot this PR is still open. Reverted back, only relevant code appears.

**Comment by adrukh** [comment]
Thanks 👍

**Pull request merged**